campaignion_newsletters/src/Subscription.php

```diff
@@ -31,18 +44,36 @@ public static function byData($list_id, $email) {
       ->execute();
     if ($row = $result->fetch()) {
       return new static($row, FALSE);
-    } else {
+    }
+    else {
       return static::fromData($list_id, $email);
     }
   }
 
+  /**
+   * Construct a new instance from data.
+   *
+   * @param string $list_id
+   *   A list id.
+   * @param string $email
+   *   An email address.
+   */
   public static function fromData($list_id, $email) {
     return new static(array(
       'list_id' => $list_id,
       'email' => $email,
     ), TRUE);
   }
 
+  /**
+   * Construct a new instance from email addresses found in the DB.
```

Comment:
That should be "Load all subscriptions for one email address." or something similar.